core/src/main/java/org/apache/carbondata/core/carbon/datastore/SegmentTaskIndexStore.java

```diff
@@ -185,15 +210,25 @@ private SegmentTaskIndexWrapper loadAndGetTaskIdToSegmentsMap(
           // acquire lock to lod the segment
           synchronized (segmentLoderLockObject) {
             segmentTaskIndexWrapper = (SegmentTaskIndexWrapper) lruCache.get(lruCacheKey);
-            if (null == segmentTaskIndexWrapper) {
-              // creating a map of take if to table segment
-              taskIdToSegmentIndexMap = new HashMap<TaskBucketHolder, AbstractIndex>();
-              segmentTaskIndexWrapper = new SegmentTaskIndexWrapper(taskIdToSegmentIndexMap);
+            if (null == segmentTaskIndexWrapper || tableSegmentUniqueIdentifier
+                .isSegmentUpdated()) {
+              // if the segment is updated then get the existing block task id map details
+              // so that the same can be updated after loading the btree.
+              if (tableSegmentUniqueIdentifier.isSegmentUpdated()
+                  && null != segmentTaskIndexWrapper) {
```

Comment:
double check for tableSegmentUniqueIdentifier.isSegmentUpdated() in above if block and this if block. Please remove one of them